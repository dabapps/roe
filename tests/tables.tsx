import * as React from 'react';
import * as renderer from 'react-test-renderer';

import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableHeader,
  TableRow
} from '../src/ts/';

describe('Row', () => {

  it('should match snapshot', () => {
    const tree = renderer.create(
      <Table>
        <TableHead>
          <TableRow>
            <TableHeader>
              Header
            </TableHeader>
          </TableRow>
        </TableHead>
        <TableBody>
          <TableRow>
            <TableCell>
              Cell
            </TableCell>
          </TableRow>
        </TableBody>
      </Table>
    );

    expect(tree).toMatchSnapshot();
  });

  describe('Table', () => {

    it('should set the collapse size', () => {
      const tree = renderer.create(
        <Table collapse="xs" />
      );

      expect(tree).toMatchSnapshot();
    });

    it('should fix row headers to specified width', () => {
      const tree = renderer.create(
        <Table fixRowHeaders rowHeaderWidth={100}>
          <TableHead>
            <TableRow>
              <TableHeader width={100}>
                Header
              </TableHeader>
            </TableRow>
          </TableHead>
          <TableBody>
            <TableRow>
              <TableCell width={100}>
                Cell
              </TableCell>
            </TableRow>
          </TableBody>
        </Table>
      );

      expect(tree).toMatchSnapshot();
    });

    it('should apply table styles classes', () => {
      const tree = renderer.create(
        <Table striped bordered condensed hover />
      );

      expect(tree).toMatchSnapshot();
    });

    it('should apply some boolean classes', () => {
      const tree = renderer.create(
        <Table
          striped
          bordered
          hover
          condensed
          fill
          fixed
        />
      );

      expect(tree).toMatchSnapshot();
    });

    it('should allow disabling scrolling', () => {
      const tree = renderer.create(
        <Table
          scrollable={false}
        />
      );

      expect(tree).toMatchSnapshot();
    });

    it('should use a component prop as the table component', () => {
      const tree = renderer.create(
        <Table
          component="p"
        />
      );

      expect(tree).toMatchSnapshot();
    });

  });

  describe('TableHeader', () => {

    it('should render a non breaking space if no children provided', () => {
      const tree = renderer.create(
        <TableHeader />
      );

      expect(tree).toMatchSnapshot();
    });

    it('should render the number 0', () => {
      expect(renderer.create(
        <TableHeader>{0}</TableHeader>
      )).toMatchSnapshot();
    });

    it('should not render null, false, undefined or empty string', () => {
      expect(renderer.create(
        <TableHeader>{null}</TableHeader>
      )).toMatchSnapshot();

      expect(renderer.create(
        <TableHeader>{false}</TableHeader>
      )).toMatchSnapshot();

      expect(renderer.create(
        <TableHeader>{undefined}</TableHeader>
      )).toMatchSnapshot();

      expect(renderer.create(
        <TableHeader>{''}</TableHeader>
      )).toMatchSnapshot();
    });

  });

  describe('TableCell', () => {

    it('should render a non breaking space if no children provided', () => {
      const tree = renderer.create(
        <TableCell />
      );

      expect(tree).toMatchSnapshot();
    });

    it('should render the number 0', () => {
      expect(renderer.create(
        <TableCell>{0}</TableCell>
      )).toMatchSnapshot();
    });

    it('should not render null, false, undefined or empty string', () => {
      expect(renderer.create(
        <TableCell>{null}</TableCell>
      )).toMatchSnapshot();

      expect(renderer.create(
        <TableCell>{false}</TableCell>
      )).toMatchSnapshot();

      expect(renderer.create(
        <TableCell>{undefined}</TableCell>
      )).toMatchSnapshot();

      expect(renderer.create(
        <TableCell>{''}</TableCell>
      )).toMatchSnapshot();
    });

  });

});
