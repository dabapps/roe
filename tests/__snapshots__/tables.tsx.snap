// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Table should allow disabling scrolling 1`] = `
<div
  className="table-wrapper"
>
  <div
    style={
      Object {
        "paddingLeft": null,
      }
    }
  >
    <div
      className={undefined}
    >
      <table
        className="table sm-collapse"
      />
    </div>
  </div>
</div>
`;

exports[`Row Table should apply some boolean classes 1`] = `
<div
  className="table-wrapper"
>
  <div
    style={
      Object {
        "paddingLeft": null,
      }
    }
  >
    <div
      className="table-scroller"
    >
      <table
        className="table sm-collapse striped bordered hover condensed fill fixed"
      />
    </div>
  </div>
</div>
`;

exports[`Row Table should apply table styles classes 1`] = `
<div
  className="table-wrapper"
>
  <div
    style={
      Object {
        "paddingLeft": null,
      }
    }
  >
    <div
      className="table-scroller"
    >
      <table
        className="table sm-collapse striped bordered hover condensed"
      />
    </div>
  </div>
</div>
`;

exports[`Row Table should fix row headers to specified width 1`] = `
<div
  className="table-wrapper"
>
  <div
    style={
      Object {
        "paddingLeft": 100,
      }
    }
  >
    <div
      className="table-scroller"
    >
      <table
        className="table sm-collapse fix-column-headers"
      >
        <thead
          className="table-head"
        >
          <tr
            className="table-row"
          >
            <th
              className="table-header"
              style={
                Object {
                  "maxWidth": 100,
                  "width": 100,
                }
              }
            >
              Header
            </th>
          </tr>
        </thead>
        <tbody
          className="table-body"
        >
          <tr
            className="table-row"
          >
            <td
              className="table-cell"
              style={
                Object {
                  "maxWidth": 100,
                  "width": 100,
                }
              }
            >
              Cell
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Row Table should set the collapse size 1`] = `
<div
  className="table-wrapper"
>
  <div
    style={
      Object {
        "paddingLeft": null,
      }
    }
  >
    <div
      className="table-scroller"
    >
      <table
        className="table xs-collapse"
      />
    </div>
  </div>
</div>
`;

exports[`Row Table should use a component prop as the table component 1`] = `
<div
  className="table-wrapper"
>
  <div
    style={
      Object {
        "paddingLeft": null,
      }
    }
  >
    <div
      className="table-scroller"
    >
      <p
        className="table sm-collapse"
      />
    </div>
  </div>
</div>
`;

exports[`Row TableCell should not render null, false, undefined or empty string 1`] = `
<td
  className="table-cell"
  style={
    Object {
      "maxWidth": undefined,
      "width": undefined,
    }
  }
>
   
</td>
`;

exports[`Row TableCell should not render null, false, undefined or empty string 2`] = `
<td
  className="table-cell"
  style={
    Object {
      "maxWidth": undefined,
      "width": undefined,
    }
  }
>
   
</td>
`;

exports[`Row TableCell should not render null, false, undefined or empty string 3`] = `
<td
  className="table-cell"
  style={
    Object {
      "maxWidth": undefined,
      "width": undefined,
    }
  }
>
   
</td>
`;

exports[`Row TableCell should not render null, false, undefined or empty string 4`] = `
<td
  className="table-cell"
  style={
    Object {
      "maxWidth": undefined,
      "width": undefined,
    }
  }
>
   
</td>
`;

exports[`Row TableCell should render a non breaking space if no children provided 1`] = `
<td
  className="table-cell"
  style={
    Object {
      "maxWidth": undefined,
      "width": undefined,
    }
  }
>
   
</td>
`;

exports[`Row TableCell should render the number 0 1`] = `
<td
  className="table-cell"
  style={
    Object {
      "maxWidth": undefined,
      "width": undefined,
    }
  }
>
  0
</td>
`;

exports[`Row TableHeader should not render null, false, undefined or empty string 1`] = `
<th
  className="table-header"
  style={
    Object {
      "maxWidth": undefined,
      "width": undefined,
    }
  }
>
   
</th>
`;

exports[`Row TableHeader should not render null, false, undefined or empty string 2`] = `
<th
  className="table-header"
  style={
    Object {
      "maxWidth": undefined,
      "width": undefined,
    }
  }
>
   
</th>
`;

exports[`Row TableHeader should not render null, false, undefined or empty string 3`] = `
<th
  className="table-header"
  style={
    Object {
      "maxWidth": undefined,
      "width": undefined,
    }
  }
>
   
</th>
`;

exports[`Row TableHeader should not render null, false, undefined or empty string 4`] = `
<th
  className="table-header"
  style={
    Object {
      "maxWidth": undefined,
      "width": undefined,
    }
  }
>
   
</th>
`;

exports[`Row TableHeader should render a non breaking space if no children provided 1`] = `
<th
  className="table-header"
  style={
    Object {
      "maxWidth": undefined,
      "width": undefined,
    }
  }
>
   
</th>
`;

exports[`Row TableHeader should render the number 0 1`] = `
<th
  className="table-header"
  style={
    Object {
      "maxWidth": undefined,
      "width": undefined,
    }
  }
>
  0
</th>
`;

exports[`Row should match snapshot 1`] = `
<div
  className="table-wrapper"
>
  <div
    style={
      Object {
        "paddingLeft": null,
      }
    }
  >
    <div
      className="table-scroller"
    >
      <table
        className="table sm-collapse"
      >
        <thead
          className="table-head"
        >
          <tr
            className="table-row"
          >
            <th
              className="table-header"
              style={
                Object {
                  "maxWidth": undefined,
                  "width": undefined,
                }
              }
            >
              Header
            </th>
          </tr>
        </thead>
        <tbody
          className="table-body"
        >
          <tr
            className="table-row"
          >
            <td
              className="table-cell"
              style={
                Object {
                  "maxWidth": undefined,
                  "width": undefined,
                }
              }
            >
              Cell
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
